lib/block_view.go

```diff
@@ -6881,6 +6956,22 @@ func (bav *UtxoView) _connectUpdateNFT(
 			txMeta.NFTPostHash.String())
 	}
 
+	if isBuyNow && blockHeight < BuyNowNFTBlockHeight {
+		return 0, 0, nil, errors.Wrapf(RuleErrorBuyNowNFTBeforeBlockHeight, "_connectUpdateNFT: ")
+	}
+
+	// We only need to make these checks if we are past the Buy Now NFT Block Height
+	if blockHeight >= BuyNowNFTBlockHeight {
```

Comment:
Nit: It seems better to reduce complexity and just run these checks regardless of the block height.